src/iosrv/init.c

```diff
@@ -354,7 +368,10 @@ server_init(int argc, char *argv[])
 	D_INFO("Module interface successfully initialized\n");
 
 	/* initialize the network layer */
-	if (sys_map_path != NULL)
+	d_getenv_bool("DAOS_PMIXLESS", &pmixless);
+	if (getenv("PMIX_RANK") == NULL)
```

Comment:
looks a little confuse for above two check.
or should it be "if (pmixless && getenv("PMIX_RANK") == NULL)"?
or with "if (pmixless && getenv("PMIX_RANK") != NULL) return err"?
or some comments is helpful, seems DAOS_PMIXLESS is not must needed.

Reply:
Thanks, @liuxuezhao. I had three cases in mind:

1. Current usage (PMIX_RANK && !DAOS_PMIXLESS): use PMIx;
2. No orterun (!PMIX_RANK): don't use PMIx;
3. With orterun but don't want PMIx (PMIX_RANK && DAOS_PMIXLESS): don't use PMIx.

In other words, if !PMIX_RANK, we have to avoid using PMIx, else we look at DAOS_PMIXLESS.

Reply:
Eventually, I think we'll just remove the PMIX_RANK check and DAOS_PMIXLESS.